src/program/lwaftr/query/selftest.sh

```diff
@@ -51,35 +58,29 @@ function fatal {
 }
 
 function test_lwaftr_query {
-    local pid=$1
-    # FIXME: Sometimes lwaftr query gets stalled. Add timeout.
-    local lineno=`timeout 1 ./snabb lwaftr query $pid | wc -l`
+    ./snabb lwaftr query $@ > $TEMP_FILE
+    local lineno=`cat $TEMP_FILE | wc -l`
     if [[ $lineno -gt 1 ]]; then
-        echo "Success: lwaftr query $pid"
+        echo "Success: lwaftr query $@"
     else
-        fatal "lwaftr query $pid"
-    fi
-}
-
-function test_lwaftr_query_filter {
-    local pid=$1
-    local filter=$2
-    local lineno=`timeout 1 ./snabb lwaftr query $pid $filter | wc -l`
-    if [[ $lineno -gt 1 ]]; then
-        echo "Success: lwaftr query $pid $filter"
-    else
-        fatal "lwaftr query $pid"
+        fatal "lwaftr query $@"
     fi
 }
 
 # Run lwAFTR.
-tmux_launch "lwaftr" "./snabb lwaftr run --reconfigurable --conf $LWAFTR_CONF --v4 $SNABB_PCI0 --v6 $SNABB_PCI1"
+tmux_launch "lwaftr" "./snabb lwaftr run --reconfigurable --name lwaftr --conf $LWAFTR_CONF --v4 $SNABB_PCI0 --v6 $SNABB_PCI1" "lwaftr.log"
 sleep 2
 
-# Run tests.
-pid=$(get_lwaftr_instance)
+# Test query all.
+test_lwaftr_query -l
+
+# Test query by pid.
```

Comment:
That would -also- be good - it covers 2 important code paths that this approach cannot - but I mean given that you're running it with reconfigurable, to get counters from -both- the leader and the follower pid (with a test each), not just one or the other.

(A sketch of what I mean):
```
test_lwaftr_query follower_pid memuse-ip
test_lwaftr_query leader_pid memuse-ip
```